plugins/node/instrumentation-typeorm/src/typeorm.ts

```diff
@@ -0,0 +1,471 @@
+          ? contextWithSpan
+          : suppressTypeormInternalTracing(contextWithSpan);
+
+        const contextWithSuppressTracing = self._config
+          .suppressInternalInstrumentation
+          ? suppressTracing(traceContext)
+          : traceContext;
+
+        return context.with(contextWithSuppressTracing, () =>
+          self._endSpan(() => original.apply(this, args), span)
+        );
+      };
+    };
+  }
+
+  private _patchQueryBuilder() {
+    const self = this;
+    return (original: Function) => {
+      return function (
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        this: typeorm.SelectQueryBuilder<any>,
+        ...args: unknown[]
+      ) {
+        if (isTypeormInternalTracingSuppressed(context.active())) {
+          return original.apply(this, args);
+        }
+        const sql = this.getQuery();
+        const parameters = this.getParameters();
+        const mainTableName = this.getMainTableName();
+        const operation = this.expressionMap.queryType;
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const connectionOptions: any = this.connection?.options;
+        const attributes: Record<string, any> = {
+          [SEMATTRS_DB_SYSTEM]: connectionOptions.type,
+          [SEMATTRS_DB_USER]: connectionOptions.username,
+          [SEMATTRS_NET_PEER_NAME]: connectionOptions.host,
+          [SEMATTRS_NET_PEER_PORT]: connectionOptions.port,
+          [SEMATTRS_DB_NAME]: connectionOptions.database,
+          [SEMATTRS_DB_OPERATION]: operation,
+          [SEMATTRS_DB_STATEMENT]: sql,
+          [SEMATTRS_DB_SQL_TABLE]: mainTableName,
+        };
+        if (self._config.collectParameters) {
+          try {
+            attributes[ExtendedDatabaseAttribute.DB_STATEMENT_PARAMETERS] =
+              JSON.stringify(parameters);
+          } catch (err) {
+            /* */
+          }
+        }
+        const span: Span = self.tracer.startSpan(
+          `TypeORM ${operation} ${mainTableName}`,
+          {
+            kind: SpanKind.CLIENT,
+            attributes,
+          }
+        );
+
+        const contextWithSpan = trace.setSpan(context.active(), span);
+
+        const traceContext = self._config.enableInternalInstrumentation
+          ? contextWithSpan
+          : suppressTypeormInternalTracing(contextWithSpan);
+
+        const contextWithSuppressTracing = self._config
+          ?.suppressInternalInstrumentation
+          ? suppressTracing(traceContext)
+          : traceContext;
+
+        return context.with(contextWithSuppressTracing, () =>
+          self._endSpan(() => original.apply(this, args), span)
+        );
+      };
+    };
+  }
+
+  private getOperationName(statement: string) {
+    let operation = 'raw query';
+    if (typeof statement === 'string') {
+      statement = statement.trim();
+      try {
+        operation = statement.split(' ')[0].toUpperCase();
```

Comment:
[The semantic conventions for database](https://opentelemetry.io/docs/specs/semconv/database/database-spans/#common-attributes) says:

> [2]: When setting this to an SQL keyword, it is not recommended to attempt any client-side parsing of db.statement just to get this property, but it should be set if the operation name is provided by the library being instrumented

Since this code does not cover many valid cases, can introduce performance hit, and the semantic conventions specifically advise aginst it, I would consider removing it from the instrumentation at the moment, or introducing a config option to opt-in to it.

If we end up keeping it, we can consider refactoring it for better performance - so that we don't need to parse the entire string and create an array just to get the first word:
```
let firstSpaceIndex = statement.indexOf(' ');
operation = firstSpaceIndex === -1 ? statement.toUpperCase() : statement.substring(0, firstSpaceIndex).toUpperCase();
```

Reply:
Removed it completely for now